Keep cairo native behind a feature flag

cptartur opened the pull request:
<!-- Reference any GitHub issues resolved by this PR -->

Closes #

## Introduced changes

<!-- A brief description of the changes -->

-

## Checklist

<!-- Make sure all of these are complete -->

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

cptartur committed:
Keep cairo native behind a feature flag

franciszekjob commented on crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/declare.rs:
```diff
@@ -73,11 +74,14 @@ fn get_contract_class(contract_artifact: &StarknetContractArtifacts) -> Runnable
     .expect("Failed to read contract class from json");
 
-    match &contract_artifact.executor {
+    #[cfg(feature = "cairo-native")]
+    return match &contract_artifact.executor {
         None => RunnableCompiledClass::V1(contract_class),
         Some(executor) => RunnableCompiledClass::V1Native(NativeCompiledClassV1::new(
```
Can't this be as follows?
```suggestion
    return match &contract_artifact.executor {
        None => RunnableCompiledClass::V1(contract_class),
        #[cfg(feature = "cairo-native")]
        Some(executor) => RunnableCompiledClass::V1Native(NativeCompiledClassV1::new(
```

cptartur replied:
It cannot because `executor` field itself is behind a feature flag as well.

franciszekjob commented on crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs:
```diff
@@ -185,7 +187,6 @@ pub fn execute_call_entry_point(
                     context,
                 )
             } else {
-                println!("Executing native entry point");
```
nit: pls remove this in target branch (`2714-cairo-native`)

cptartur replied:
Don't want to rebase the whole stack 😅

franciszekjob commented on crates/forge/src/profile_validation/backtrace.rs:
```diff
@@ -7,17 +7,20 @@ use semver::Version;
 
 /// Checks if backtrace can be generated based on scarb version, profile settings extracted from
 /// the provided [`Metadata`] and if native execution is disabled in the provided [`TestArgs`].
+#[allow(unused_variables)]
```
Is this needed?

ksew1 commented on crates/scarb-api/src/artifacts.rs:
```diff
@@ -20,23 +21,32 @@ pub struct StarknetContractArtifacts {
     fn eq(&self, other: &Self) -> bool {
-        self.sierra == other.sierra
-            && self.casm == other.casm
+        let eq = self.sierra == other.sierra && self.casm == other.casm;
```
Wouldn't this be cleaner

```rs
let eq = self.sierra == other.sierra && self.casm == other.casm;

#[cfg(feature = "cairo-native")]
let eq = eq && self.executor.is_some() == other.executor.is_some();

eq
```

ksew1 commented on crates/sncast/src/helpers/artifacts.rs:
```diff
@@ -0,0 +1,8 @@
+/// Contains compiled Starknet artifacts
+#[derive(Debug, Clone)]
+pub struct CastStarknetContractArtifacts {
```
Why we add this?

cptartur committed:
Run native as a separate job (#3746)

<!-- Reference any GitHub issues resolved by this PR -->

Closes #

## Introduced changes

<!-- A brief description of the changes -->

-

## Checklist

<!-- Make sure all of these are complete -->

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`
